src/formatDate.js

```diff
@@ -8,7 +8,70 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat[fromFormat.length - 1];
+  const newSeparator = toFormat[toFormat.length - 1];
+  const dateParts = date.split(oldSeparator);
+  let year = '';
+  let month = '';
+  let day = '';
+  const formatDateParts = [];
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    switch (fromFormat[i][0]) {
+      case 'Y':
+        year = dateParts[i];
+        break;
+      case 'M':
+        month = dateParts[i];
+        break;
+      case 'D':
+        day = dateParts[i];
+        break;
+      default:
+        throw new Error('not correct format');
+    }
+  }
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    switch (toFormat[i][0]) {
+      case 'Y':
+        formatDateParts.push(getFormatYear(year, toFormat[i]));
+        break;
+      case 'M':
+        formatDateParts.push(month);
+        break;
+      case 'D':
+        formatDateParts.push(day);
+        break;
+      default:
+        throw new Error('not correct format');
```

Comment:
Similar to the previous comment, the error message here could be more descriptive. Indicating which part of the format is incorrect and why can greatly improve the debugging experience for users.